java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java

```diff
@@ -253,28 +255,25 @@ public static void subscribeMinionToChannels(SystemQuery systemQuery, MinionServ
                         // ActivationKey without base channel (SUSE Manager Default)
                         () -> {
                             Set<SUSEProduct> suseProducts = identifyProduct(systemQuery, server, grains);
-                            return findChannelsForProducts(suseProducts, minionId);
+                            Set<Channel> channelsForProducts = findChannelsForProducts(suseProducts, minionId);
+                            Optional<Channel> baseChannel = channelsForProducts.stream()
+                                    .filter(c -> c.isBaseChannel()).findFirst();
+                            if (baseChannel.isPresent()) {
+                                // assign the identified base channel and activation key channels related
+                                // to that base channel only
+                                return Stream.concat(
+                                        Stream.of(baseChannel.get()),
+                                        ak.getChannels().stream()
+                                                .filter(c -> c.getParentChannel().getId()
+                                                        .equals(baseChannel.get().getId())))
+                                        .collect(toSet());
```

Comment:
And when the customer did not select any child channel?
He just set "SUSE Manager Default" and no child channel selection. Should we act as there is no activation key set at all? I think this was the idea.

Reply:
I see, but actually it becomes tricky to understand what is the right option to choose. If one can use an AK with "SUSE Manager Default" and no child selection, or at the same time he can also use no AK and achieving the same result, what's the reason for having two completely different approaches? From my POV it is much simpler to understand as the following:
- no AK? let SUSE Manager do its homework for you
- AK used but "SUSE Manager Default"? I tell SUSE Manager that I don't really know the base product on the server I am bootstrapping, but I know the set of channels I want in case of one or the other

Reply:
Do not look only at Software Channels. AK can do much more.

But ending up with a system which does not have channel was in the past "forbidden".
For several reason we allowed it. But I think whenever possible we should put software channels to a system.